### ISSUE TITLE
cppbuild task outputs executable file in the wrong directory

### ISSUE THREAD

**Issue by 8549** [microsoft/vscode#109957]
- VSCode Version: 1.50.1 (user setup)
- OS Version: Windows_NT x64 10.0.20246

Steps to Reproduce:

1. Setup `tasks.json` as shown in picture
2. With `2.c` as the currently open file, Terminal > Run Build Task
3. Executable is created in workspace folder (`infoA`), not in subdirectory (`lab6`)

![image](https://user-images.githubusercontent.com/15671077/98117152-969b5280-1ea9-11eb-8862-a01398f69824.png)

Does this issue occur when all extensions are disabled?: Cannot test without C++ extension

I wrote a little echo task to check the content of VS Codes variables:
```yaml
{
  "type": "shell",
  "label": "Test vars",
  "command": "echo",
  "args": [
    "${fileDirname}/${fileBasenameNoExtension}",
    " - ",
    "${workspaceFolder}"
  ],
  "problemMatcher": []
}
```
and the output is:
```shell
> Executing task: echo /home/filippo/infoA/lab6/2 ' - ' /home/filippo/infoA <

/home/filippo/infoA/lab6/2  -  /home/filippo/infoA
```
From this task it seems the variables are correct, but something goes wrong in the compilation.

**Comment by 8549** [comment]
I don't think it is caused by an extension. Unfortunately i cannot reproduce it disabling ALL extensions as i'm developing on WSL2. I disabled all third party extensions and was left only with

- C/C++ (Locally installed)
- Remote - WSL (Globally installed)
- Remote Development (Globally installed)

But the same issue persists. Is there any way to find which of these extensions is the culprit? Or is it really a VS Code bug?

**Comment by alexr00** [comment]
It would be the C++ extension since your task is of type `cppbuild`.

**Comment by 8549** [comment]
It makes sense, and i found the corresponding issue in https://github.com/microsoft/vscode-cpptools/issues/6366. Thanks for your help.

**Comment by sean-mcmanus** [comment]
It's actually not bug https://github.com/microsoft/vscode-cpptools/issues/6366 (which got fixed), but bug microsoft/vscode-cpptools#6386 with the C/C++ extension -- we're looking at shipping a patch update tomorrow.

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.1.2
